### ISSUE TITLE
rebindLocalRewriteRules re-adds global rewrite rules

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#2433]
See my comment at commit https://github.com/agda/agda/commit/943e9bee0a1fc58575b8e292e28d03c0ba98bbc8

**Comment by andreasabel** [comment]
After my patch, one can reproduce the issue e.g. with this testcase:
```agda
-- Andreas, 2017-01-26, issue #2433
-- Provoke a crash if Agda thinks we have local rewrite rules in this file

{-# OPTIONS --rewriting #-}
{-# OPTIONS -v rewriting.local.crash:1000 #-}

open import Common.Equality
open import Common.Nat

{-# BUILTIN REWRITE _≡_ #-}

plus-0 : ∀ x → x + 0 ≡ x
plus-0 zero    = refl
plus-0 (suc x) = cong suc (plus-0 x)

{-# REWRITE plus-0 #-}

plus-suc : ∀ x y → x + suc y ≡ suc (x + y)
plus-suc zero    y = refl
plus-suc (suc x) y = cong suc (plus-suc x y)

-- Error:
-- We reached a program point we did not want to reach.
-- Location of the error: src/full/Agda/TypeChecking/Rewriting.hs:312

-- Should not crash.
```

**Comment by jespercockx** [comment]
Fixed by 46b4a0f87869b5c9bc847e43cbfbc7a4135361c9
